selfservice/strategy/oidc/provider_microsoft.go

```diff
@@ -67,7 +69,49 @@ func (m *ProviderMicrosoft) Claims(ctx context.Context, exchange *oauth2.Token)
 		return nil, errors.WithStack(herodot.ErrInternalServerError.WithReasonf("Unable to initialize OpenID Connect Provider: %s", err))
 	}
 
-	return m.verifyAndDecodeClaimsWithProvider(ctx, p, raw)
+	claims, err := m.verifyAndDecodeClaimsWithProvider(ctx, p, raw)
+	if err != nil {
+		return nil, err
+	}
+
+	return m.updateSubject(ctx, claims, exchange)
+}
+
+func (m *ProviderMicrosoft) updateSubject(ctx context.Context, claims *Claims, exchange *oauth2.Token) (*Claims, error) {
+	if m.config.SubjectSource == "me" {
+		o, err := m.OAuth2(ctx)
+		if err != nil {
+			return nil, errors.WithStack(herodot.ErrInternalServerError.WithReasonf("%s", err))
+		}
+
+		client := o.Client(ctx, exchange)
+
+		u, err := url.Parse("https://graph.microsoft.com/v1.0/me")
+		if err != nil {
+			return nil, errors.WithStack(herodot.ErrInternalServerError.WithReasonf("%s", err))
+		}
+		req, err := http.NewRequest("GET", u.String(), nil)
+		if err != nil {
+			return nil, errors.WithStack(herodot.ErrInternalServerError.WithReasonf("%s", err))
+		}
+
+		resp, err := client.Do(req)
+		if err != nil {
+			return nil, errors.WithStack(herodot.ErrInternalServerError.WithReasonf("%s", err))
+		}
+		defer resp.Body.Close()
+
+		var user struct {
+			Id string `json:"id"`
+		}
+		if err := json.NewDecoder(resp.Body).Decode(&user); err != nil {
+			return nil, errors.WithStack(herodot.ErrInternalServerError.WithReasonf("%s", err))
+		}
+
+		claims.Subject = user.Id
```

Comment:
Is the value from the `me` endpoint different than the one from the userinfo endpoint?

Reply:
Yes, they are different. Have checked this with my own account.

Reply:
Could you share what their formats look like? And is there no way to get this other ID from the userinfo endpoint?

Reply:
Using an access token with scope [User.Read openid profile email] I receive

- from /userinfo

{
    "sub": "AAAAAAAAAAAAAAAAAAAAAENMusLqo_f5vKwPzixWdNS",
    "email": "some@mail.domain",
    "picture": "https://graph.microsoft.com/v1.0/me/photo/$value"
}

- from /me

{
    "@odata.context": "https://graph.microsoft.com/v1.0/$metadata#users/$entity",
    "displayName": "",
    "surname": null,
    "givenName": null,
    "id": "7wt4391vs271486n",
    "userPrincipalName": "some@mail.domain",
    "businessPhones": [],
    "jobTitle": null,
    "mail": null,
    "mobilePhone": null,
    "officeLocation": null,
    "preferredLanguage": null
}

Reply:
Thank you, this helps!